website/docs/cli/configuration/configuration.mdx

```diff
@@ -24,15 +24,58 @@ Think of this file as where you [bootstrap the settings or configuration of your
 you'd specify the command to run (e.g. [`opentofu`](/core-concepts/projects/configuration/opentofu)),
 the base path location of the components, and so forth.
+## Base Path
 
-## Configuration File (`atmos.yaml`)
+The base path defines the root directory for components, stacks, workflows, and Atmos configurations. It can be set using the `ATMOS_BASE_PATH` environment variable or by passing the `--base-path` command-line argument. If both are set, the `--base-path` will take precedence. Both absolute and relative paths are supported. When a relative path is provided, it will be expanded to an absolute path relative to the current working directory, as Atmos executes commands that may change the working directory (e.g., Terraform).
+
+When the `base_path` is explicitly defined, it will always be resolved to the absolute path relative to the current working directory, if an absolute path was not already set.
+
+When the `base_path` is not explicitly provided or is empty, Atmos will attempt to infer the appropriate path using the following strategy, in order of precedence:
+1. If the current working directory contains any of the following files or directories: `atmos.yaml`, `.atmos.yaml`, `.github/atmos.yaml`, or `atmos.d/**/*`, Atmos will use the directory containing these files as the `base_path`.
+2. If a Git repository is detected, Atmos uses the root of the Git repository as the `base_path`.
+3. If neither of the above applies, Atmos defaults to the absolute path of `./` as the `base_path`.
 
-The CLI config is loaded from the following locations (from lowest to highest priority):
+Relative paths, such as `components.terraform.base_path`, `components.helmfile.base_path`, `stacks.base_path`, `workflows.base_path`, `schemas.jsonschema.base_path`, and `schemas.opa.base_path`, are resolved relative to the specified `base_path`. Absolute paths for these settings will be respected and used as-is. 
 
-- System directory (`/usr/local/etc/atmos/atmos.yaml` on Linux, `%LOCALAPPDATA%/atmos/atmos.yaml` on Windows)
+Each setting (`components.terraform.base_path`, `components.helmfile.base_path`, `stacks.base_path`, and `workflows.base_path`) operates independently and can accept either an absolute or a relative path.
+
+## Configuration File (`atmos.yaml`)
+
+The Atmos (CLI) configuration is loaded from the following locations, in order of precedence, and deep-merged: configurations listed earlier have higher priority, meaning they override any conflicting sections defined in subsequent locations:
+- The `--config` parameter is the relative or absolute path to a valid configuration file or directory containing strictly atmos configuration file(s). It has the highest precedence, meaning it supersedes everything.
+- The `$XDG_CONFIG_HOME/atmos/atmos.yaml` file (if `$XDG_CONFIG_HOME` is set) allows the user to specify their preferences and overrides. It is only superseded by the `--config` flag.
```

Comment:
Let's see if there's a go XDG library so we support conventional usage.
https://github.com/adrg/xdg

Reply:
Note, that `viper` should already support XDG, but we need to verify this.